warehouse/query-core/src/main/java/datawave/query/jexl/JexlASTHelper.java

```diff
@@ -594,29 +594,32 @@ public static String deconstructIdentifier(String fieldName) {
     }
     
     /**
-     * Remove the {@link #IDENTIFIER_PREFIX} from the beginning of a fieldName if it exists
-     * 
+     * Remove the {@link #IDENTIFIER_PREFIX} from the beginning of a fieldName if it exists Remove the {@link #GROUPING_CHARACTER_SEPARATOR} from the end of a
+     * fieldName if it exists
+     *
      * @param fieldName
      * @param includeGroupingContext
      * @return
      */
     public static String deconstructIdentifier(String fieldName, boolean includeGroupingContext) {
         if (fieldName != null && fieldName.length() > 1) {
             if (!includeGroupingContext) {
-                fieldName = removeGroupingContext(fieldName);
-            }
-            
-            if (fieldName.charAt(0) == IDENTIFIER_PREFIX) {
-                return fieldName.substring(1);
+                int groupingOffset = fieldName.indexOf(GROUPING_CHARACTER_SEPARATOR);
+                if (-1 != groupingOffset) {
+                    if (fieldName.charAt(0) == IDENTIFIER_PREFIX) {
+                        return new String(fieldName.getBytes(), 1, groupingOffset - 1);
+                    }
+                    return new String(fieldName.getBytes(), 0, groupingOffset);
+                }
             }
         }
         
-        return fieldName;
+        return (fieldName.charAt(0) == IDENTIFIER_PREFIX) ? fieldName.substring(1) : fieldName;
     }
     
     /**
      * Rebuild the identifier with the {@link #IDENTIFIER_PREFIX} if the identifier starts with an invalid character per the Jexl IDENTIFIER definition
-     * 
+     *
```

Comment:
The final line can still be `return fieldName`. You can simplify the body of the method to be along these lines (just pseudo code, I don't expect this to run)
```
int startIndex = fieldName.charAt(0) == IDENTIFIER_PREFIX ? 1 : 0;
int groupingOffset = fieldName.indexOf(GROUPING_CHAR_SEP)
int stopIndex = groupingOffset == -1 ? fieldName.length()-1 : groupingOffset

fieldName = new String(fieldName.getBytes, startIndex, stopIndex)
```

Reply:
Tests should be fairly straightforward as well
```
FIELD => FIELD
$FIELD => FIELD

# when includeGroupingContext is false
FIELD.1.2 => FIELD
$FIELD.1.2 => FIELD

# when includeGroupingContext is true
FIELD.1.2 => FIELD.1.2
$FIELD.1.2 => FIELD.1.2
```

Reply:
Updated method body as suggested and added unit tests.